tests/python/test-parameter-server.py

```diff
@@ -9,360 +10,56 @@
-      so that ros and opencv can operate on the camera frame correctly -->\
-  <joint name=\"camera_optical_joint\" type=\"fixed\">\
-    <!-- these values have to be these values otherwise the gazebo camera image\
-        won't be aligned properly with the frame it is supposedly originating from -->\
-    <origin rpy=\"-1.57079632679 0 -1.57079632679\" xyz=\"0 0 0\"/>\
-    <parent link=\"camera_link\"/>\
-    <child link=\"camera_link_optical\"/>\
-  </joint>\
-  <link name=\"camera_link_optical\">\
-  </link>\
-  <transmission name=\"tran1\">\
-    <type>transmission_interface/SimpleTransmission</type>\
-    <joint name=\"joint1\">\
-      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>\
-    </joint>\
-    <actuator name=\"motor1\">\
-      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>\
-      <mechanicalReduction>1</mechanicalReduction>\
-    </actuator>\
-  </transmission>\
-  <transmission name=\"tran2\">\
-    <type>transmission_interface/SimpleTransmission</type>\
-    <joint name=\"joint2\">\
-      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>\
-    </joint>\
-    <actuator name=\"motor2\">\
-      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>\
-      <mechanicalReduction>1</mechanicalReduction>\
-    </actuator>\
-  </transmission>\
-</robot>\
-"
+import sys
 
-# 
-model = pin.buildModelFromXML(urdf_rrbot_model_string)
+from os.path import dirname, join, abspath
 
-print("model.nq:"+str(model.nq))
-print(len(q))
-neutral_configuration_model=pin.neutral(model)
-#print(model)
-jointIds=[]
-reduced_model=pin.buildReducedModel(model, jointIds, q)
-print(neutral_configuration_model)
-print(len(q))
+class TestParameterServer(unittest.TestCase):
 
-data = model.createData()
-com = pin.centerOfMass(model, data, q)
-pin.updateFramePlacements(model, data)
-m = data.mass[0]
-param_server.setParameter("urdf_model",urdf_rrbot_model_string)
-model2_string=param_server.getParameter("urdf_model")
-print(len(model2_string))
+    def test_set_parameter(self):
+        # Read talos model
+        path = join(dirname(dirname(abspath(__file__))),
+                    'models', 'others', 'python')
+        sys.path.append(path)
 
+        from example_robot_data.path import EXAMPLE_ROBOT_DATA_MODEL_DIR
+
+        urdf_file_name=EXAMPLE_ROBOT_DATA_MODEL_DIR+\
+           '/talos_data/robots/talos_reduced.urdf'
+
+        fs=open(urdf_file_name,'r')
+        urdf_rrbot_model_string=fs.read()
+        fs.close()
+
+        print(EXAMPLE_ROBOT_DATA_MODEL_DIR)
+        #
+        model = pin.buildModelFromXML(urdf_rrbot_model_string)
+
+        neutral_configuration_model=pin.neutral(model)
+        jointIds=[]
+        #print(model)
+        reduced_model=pin.buildReducedModel(model, jointIds,
+                                            neutral_configuration_model)
```

Comment:
If `jointIds` is empty, this does nothing. What do you want to do ?

Reply:
You are right I removed it